raiden-ts/src/raiden.ts

```diff
@@ -1437,16 +1457,29 @@ export class Raiden {
     assert(Address.is(token), [ErrorCodes.DTA_INVALID_ADDRESS, { token }], this.log.info);
     const tokenNetwork = this.state.tokens[token];
     assert(tokenNetwork, ErrorCodes.RDN_UNKNOWN_TOKEN_NETWORK, this.log.info);
-    const pfs = decode(PFS, options.pfs ?? (await this.findPFS())[0]);
+    const pfss = options.pfs ? [decode(PFS, options.pfs)] : await this.findPFS();
 
-    return await fromFetch(`${pfs.url}/api/v1/${tokenNetwork}/suggest_partner`, {
-      method: 'GET',
-      headers: { 'Content-Type': 'application/json' },
-    })
+    let firstError: Error | undefined;
+    return from(pfss)
       .pipe(
-        timeout(this.config.httpTimeout),
-        mergeMap((response) => response.text()),
-        map((text) => decode(SuggestedPartners, jsonParse(text))),
+        concatMap((pfs) =>
+          fromFetch(`${pfs.url}/api/v1/${tokenNetwork}/suggest_partner`, {
+            method: 'GET',
+            headers: { 'Content-Type': 'application/json' },
+          }).pipe(
+            timeout(this.config.httpTimeout),
+            mergeMap(async (response) => response.text()),
+            map((text) => decode(SuggestedPartners, jsonParse(text))),
+            catchError((err) => {
+              // store first error and omit to retry next pfs in list
+              if (!firstError) firstError = err;
+              return EMPTY;
+            }),
+          ),
+        ),
+        first(), // throws if no first result can be fetched/decoded
+        // if first errored, throw first seen error or pass current through
+        catchError((err) => throwError(firstError ?? err)),
```

Comment:
Just for me: how could it every happen that there is no `firstError` at this point in time/code? :thinking:

Reply:
`first` can throw if `findPFS` array is empty, so at this point in time, there was never a `firstError` on `fromFetch` from any pfs because there was none, and in this case, this will catch & rethrow the `first` error saying that the observable completed without emitting any items.
We could make it more descriptive, but I like to be transparent on non-expected errors, since they're unexpected and therefore more informative.